tests/test_ec2/test_security_groups.py

```diff
@@ -581,8 +583,8 @@ def test_sec_group_rule_limit():
     ec2_conn.authorize_security_group_egress(
         group_id=sg.id, ip_protocol="-1", src_group_id=other_sg.id
     )
-    # fill the rules up the limit
-    for i in range(1, 100):
+    # fill the rules up the limit, 49 + 1 default rule
+    for i in range(1, 49):
```

Comment:
What's the reasoning behind this change?

Reply:
Actually, As we discussed on other PR, we are no longer supporting Ec2-Classic for the SG. So by default, there will be always `SecurityGroup` in default `VPC` if no VPC is being passed while creating a SG. So for `EC2-VPC`'s SGs, one can have max of 50 Rules per SG. For egress, there will be by default 1 rule.

However, AWS is still giving backward compatibility and I haven't checked the other services in `moto` itself if they do have those EC2-Classic attributes. So should we make it default behaviour or add `TODO` for the future? thoughts?

Reply:
I think we should add TODO note because there are still many `EC2-Classic` test cases for different services which exist in terraform tests.

Reply:
Yeah, I meant the change from 100 to 50. Is it the EC2-classic rule to allow 100 SG's, or was it always max 50 (and Moto had been wrong the entire time)?

Reply:
> I think we should add TODO note because there are still many `EC2-Classic` test cases for different services which exist in terraform tests.

I'm having a difficult time to wrap my head around how Terraform would do this. Isn't EC2-classic only available/enabled on older accounts? Or is there a way to force EC2-classic behaviour by passing a specific parameter somehow

Reply:
> Yeah, I meant the change from 100 to 50. Is it the EC2-classic rule to allow 100 SG's, or was it always max 50 (and Moto had been wrong the entire time)?

- For EC2-Classic its 100 and for Ec2-VPC its 60 (We have to update it from 50 to 60).

> I'm having a difficult time to wrap my head around how Terraform would do this. Isn't EC2-classic only available/enabled on older accounts? Or is there a way to force EC2-classic behaviour by passing a specific parameter somehow

- IDK as well, they maybe have different accounts for testing purposes with EC2-Classic features or they can ask to AWS. Its Hashicorp, right? AWS can provide them 😉 . As per the [link](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-classic-platform.html), they no longer allow Ec2-Classic for the new ACC. So I guess, there is no way we can enable Ec2-classic features for our new ACCs.

Reply:
My preference would be to switch to VPC as the default behaviour, and retire any EC2-classic features as and when we encounter them.
The TF tests might fail for another year, until it's retired for good, but I expect the vast majority of people to use VPC only. So forcefully retiring our EC2-classic behaviour will improve/increase the supported features for the vast majority of Moto/Localstack users.
Any users who are still on EC2-classic will just have to pin Moto to an older version.

Reply:
Sounds good. We can blacklist those tests as we encounter them.

Reply:
Cool. Looks like we have a conflict to resolve before we can merge, but LGTM otherwise - let me know when this is ready